barretenberg/cpp/src/barretenberg/commitment_schemes_recursion/ipa_recursive.test.cpp

```diff
@@ -230,4 +230,100 @@ TEST_F(IPARecursiveTests, AccumulateMedium)
 TEST_F(IPARecursiveTests, ConstantVerifier)
 {
     test_fixed_ipa_recursive_verifier();
+}
+
+TEST_F(IPARecursiveTests, FullRecursiveVerifier)
+{
+    const size_t POLY_LENGTH = 1024;
+    Builder builder;
+    auto [stdlib_transcript, stdlib_claim] = create_ipa_claim(builder, POLY_LENGTH);
+
+    auto stdlib_pcs_vkey = std::make_shared<VerifierCommitmentKey<Curve>>(&builder, POLY_LENGTH, this->vk());
+    auto result = RecursiveIPA::full_verify_recursive(stdlib_pcs_vkey, stdlib_claim, stdlib_transcript);
+    EXPECT_TRUE(result);
+    builder.finalize_circuit(/*ensure_nonzero=*/true);
+    info("Full IPA Recursive Verifier num finalized gates for length ",
+         POLY_LENGTH,
+         " = ",
+         builder.get_num_finalized_gates());
+    EXPECT_TRUE(CircuitChecker::check(builder));
+}
+
+TEST_F(IPARecursiveTests, AccumulationAndFullRecursiveVerifier)
+{
+    const size_t POLY_LENGTH = 1024;
+
+    // We create a circuit that does two IPA verifications. However, we don't do the full verifications and instead
+    // accumulate the claims into one claim. This accumulation is done in circuit. Create two accumulators, which
+    // contain the commitment and an opening claim.
+    Builder builder;
+
+    auto [transcript_1, claim_1] = create_ipa_claim(builder, POLY_LENGTH);
+    auto [transcript_2, claim_2] = create_ipa_claim(builder, POLY_LENGTH);
+
+    // Creates two IPA accumulators and accumulators from the two claims. Also constructs the accumulated h
+    // polynomial.
+    auto [output_claim, ipa_proof] = RecursiveIPA::accumulate(this->ck(), transcript_1, claim_1, transcript_2, claim_2);
+    builder.finalize_circuit(/*ensure_nonzero=*/false);
+    info("Circuit with 2 IPA Recursive Verifiers and IPA Accumulation num finalized gates = ",
+         builder.get_num_finalized_gates());
+
+    EXPECT_TRUE(CircuitChecker::check(builder));
+
+    Builder root_rollup;
+    // Fully recursively verify this proof to check it.
+    auto stdlib_pcs_vkey =
+        std::make_shared<VerifierCommitmentKey<Curve>>(&root_rollup, 1 << CONST_ECCVM_LOG_N, this->vk());
```

Comment:
Why do we need this hard-coded size? Could you describe in an issue (you can put the comment in one of your unmerged PRs to avoid a CI run).